apiserver/controllers/controllers.go

```diff
@@ -180,6 +232,19 @@ func (a *APIController) NotFoundHandler(w http.ResponseWriter, r *http.Request)
 	json.NewEncoder(w).Encode(apiErr)
 }
 
+func (a *APIController) MetricsTokenHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
```

Comment:
We should check ```if auth.IsAdmin(ctx)```. Both here and in ```GetJWTMetricsToken()```, even if we do check that in the auth middleware. We need to guard against someone using this in a different code path and forgets to properly validate before calling these functions. Call me paranoid, but better safe than sorry.